### DISCUSSION TITLE
NRC keeps creating folders I don't want.

### DISCUSSION THREAD

**Discussion by Hipshot-zz** [Garux/netradiant-custom#188] [General]
Is there a way to hinder NRC to make folders in the game path, the preset folders. Like /valve and /assets and whatever those may contain. It's very annoying and it doesn't work to delete the folders afterwards either, they keep being recreated even if there's nothing in them.

Maybe there could be a clean-custom profile not attached to a game or some other way of modifying this?

**Comment by Garux** [comment]
If i follow right /valve and /assets are basegamename specified in .game. You normally have packs, textures etc there.
I don't think there is good reason to rework this logic. If you don't use this folder for some reason, set some different game path or change basegamename.

**Comment by Hipshot-zz** [comment]
I understand the logic and folder structure, but it's really not needed for the program to create the folders, other editors really don't, it's some sort of newb guiding not needed =) Anyway, it's what it is. I assume this is something from old Net Radiant, not created now?

So I have also tried other games, it seems that most (all?) create some sort of folder structure and even if I set another fs_game (like try to set it to blank won't work) I can't get rid of this behaviour, since it's creating sub folders too.

One would assume that the godot setting would not do this, since there's not really any fixed and set folder structure from any game.

**Comment by Garux** [comment]
Honestly idk original reasons behind this behavior, might be newb guiding, which is good, or performance aspect to not check folders existence every time.
Anyhow workarounds are different game path or basegamename.

**Comment by Hipshot-zz** [comment]
"/scripts" I can get rid of, but "/maps" I can't even with different game path or basegamename.

For scripts I just change " shaderpath="scripts"" to " shaderpath=""" That works, but there's no such thing like "mappath=" or so, maybe at least that can be added to the .game config files?

**Reply by Garux** [reply]
Idea behind different game path is to use path, where extra folder wouldn't bug you. E.g. in temp or editor folder.
I believe case of not using basegamename folder is rather exotic to craft dedicated support for.

**Comment by Hipshot-zz** [comment]
> Idea behind different game path is to use path, where extra folder wouldn't bug you. E.g. in temp or editor folder. I believe case of not using basegamename folder is rather exotic to craft dedicated support for.

Maybe I missunderstand what you mean, I have no issues directing the gamepath to like a temp folder or like the recycle bin, but then I can't read the texture folder, so I don't see how I can do this with being bugged by the extra folder. Or am I to understand that I can't solve this and should just move my maps to such a folder in the future?

I thought that the extra resource folder could be target to a texture folder, but that's not really what that's for right?

**Reply by Garux** [reply]
Apparently i do not follow what's your setup/problem then. Extra resource folder is extra VFS root.

**Reply by Hipshot-zz** [reply]
The issue is that the editor is creating the folders "assets", "maps" and "scripts" by itself, that is the issue. It keeps doing that, all the time. You can't change that, whatever you do. You can't change the basegamename, to solve this either.

At least the creation of the scripts folder can be mitigated by just changing the shaderpath line in the .game file.

**Reply by Garux** [reply]
Alright, so missing bit is info about your setup, it's unclear which problem you are solving without it.
'assets' is defined here, not in basegamename
![image](https://github.com/Garux/netradiant-custom/assets/1734117/c377e7c1-1ac6-4ef5-87f4-e1238cd2682c)

**Reply by Hipshot-zz** [reply]
See, the issue here isn't really the names of the folders, it's that they are created by the program automatically. You must have the same behaviour on your side? Let's ignore whatever the folders are called for now.

If you take a game profile, anyone, but let's say Half-life for now, and tell the editor that the game is located in "C:\NRCtest", then the program will create a "valve" folder and inside that a folder called "maps".

I'm sure this is all good for people working for Half-Life, or other games that uses this "idSoftware" base structure, like Quake or so. But if I wanted to make a game for Unity or Godot here and I have my maps in a folder called "levels" or "world" instead, then I will be left with this folder called "maps" that will always be empty and I can't get rid of it, because everytime I start NRC that folder will be created.
